src/DropAndSwapStakeRouter.sol

```diff
@@ -79,14 +79,25 @@ contract DropAndSwapStakeRouter is BaseRouter {
         _instructions.dropToken.safeTransfer(_instructions.recipient, _instructions.dropToken.balanceOf(address(this)));
     }
 
-    function rescure(ERC20 token) public {
-        // transfer dropFee to feeReceiver if possible
-        token.safeTransfer(_instructions.feeReceiver, Math.min(_instructions.dropFee, token.balanceOf(address(this))));
+    function rescue(ERC20 token, bool isGasDrop) public {
+        if (isGasDrop) {
+            // require transfer full dropFee to feeReceiver
+            if (token.balanceOf(address(this)) < _instructions.dropFee) {
+                revert("DropAndStakeByDEXUtilRouter: cannot afford drop fee");
+            }
+            token.safeTransfer(_instructions.feeReceiver, _instructions.dropFee);
+
+            // transfer all dropToken to recipient
+            _instructions.dropToken.safeTransfer(_instructions.recipient, _instructions.dropToken.balanceOf(address(this)));
+        } else {
+            // transfer all dropToken back to feeReceiver (no gas drop)
+            _instructions.dropToken.safeTransfer(_instructions.feeReceiver, _instructions.dropToken.balanceOf(address(this)));
+        }
 
-        // transfer it to recipient to avoid it stuck in this contract
+        // transfer all remainning token to recipient to avoid it stuck in this contract
         token.safeTransfer(_instructions.recipient, token.balanceOf(address(this)));
 
-        // transfer all dropToken to recipient
-        _instructions.dropToken.safeTransfer(_instructions.recipient, _instructions.dropToken.balanceOf(address(this)));
+        emit RouterDestroyed(address(this));
+        selfdestruct(payable(msg.sender));
```

Comment:
this will send all the native token to sender, which could be the factory contract?

Reply:
oh yeah that's a good catch. Although at this point router contract should have 0 balance, since we should have dropped all ACA to recipient, but let's change this to feeReceiver to be safer